scipio/src/executor.rs

```diff
@@ -293,6 +293,50 @@ impl ExecutorQueues {
     }
 }
 
+/// LocalExecutor factory, which can be used in order to configure the properties of a new
+/// [`LocalExecutor`].
+///
+/// Methods can be chained on it in order to configure it.
+///
+/// The [`spawn`] method will take ownership of the builder and create an [`io::Result`] to the
+/// [`LocalExecutor`] handle with the given configuration.
+///
+/// The [`LocalExecutor::spawn_default`] free function uses a Builder with default configuration and
+/// unwraps its return value.
+///
+/// You may want to use [`spawn`] instead of [`LocalExecutor::spawn_default`], when you want to
+/// recover from a failure to launch a thread, indeed the free function will panic where the Builder
+/// method will return a [`io::Result`].
+///
+/// # Examples
+///
+/// ```
+/// use scipio::LocalExecutorBuilder;
+///
+/// let builder = LocalExecutorBuilder::new();
+/// let ex = builder.spawn().unwrap();
+/// ```
+#[derive(Debug)]
+pub struct LocalExecutorBuilder {}
+
+impl LocalExecutorBuilder {
+    /// Generates the base configuration for spawning a [`LocalExecutor`], from which configuration
+    /// methods can be chained.
+    pub fn new() -> LocalExecutorBuilder {
+        LocalExecutorBuilder {}
+    }
+
+    /// Spawns a new [`LocalExecutor`] by taking ownership of the Builder, and returns an io::Result
+    /// to its JoinHandle.
+    pub fn spawn(&self) -> io::Result<LocalExecutor> {
```

Comment:
shouldn't this be `self` instead of `self` if you are taking ownership of the builder ?